### DISCUSSION TITLE
Where to found musl binaries?

### DISCUSSION THREAD

**Discussion by cassiofb-dev** [fastfetch-cli/fastfetch#573] [Q&A]
Hi, recently I installed termux and loved this alternative to neofetch. Lots of information, really impressive!

I'm using this [official binary](https://github.com/fastfetch-cli/fastfetch/releases/download/2.1.0/fastfetch-2.1.0-Linux.tar.gz), I'll share the logs and the system info:

## Error and LDD (seems like it needs gnu libc dependencies)

![image](https://github.com/fastfetch-cli/fastfetch/assets/61878987/4820cfb2-a8df-4bd7-8560-f1f6d214210a)

## System info (Hope to try fastfetch soon 😜)

![image](https://github.com/fastfetch-cli/fastfetch/assets/61878987/0069a30b-4b79-456c-a5d6-cf8cd8c005a7)

**Comment by cassiofb-dev** [comment]
Some time passed and it seems there are no plans for musl support. For people wanting to use it with Alpine (that uses musl) you can follow this steps: [https://wiki.alpinelinux.org/wiki/Running_glibc_programs](https://wiki.alpinelinux.org/wiki/Running_glibc_programs)

![image](https://github.com/fastfetch-cli/fastfetch/assets/61878987/4285d164-38ed-4962-ba1b-9a48f972ea0f)

**Comment by CarterLi** [comment]
I compiled fastfetch for musl from source. Please have a try

[fastfetch.gz](https://github.com/fastfetch-cli/fastfetch/files/13928891/fastfetch.gz)

**Reply by cassiofb-dev** [reply]
Thanks, it worked.  On my desktop I got this result:

---

![image](https://github.com/fastfetch-cli/fastfetch/assets/61878987/3eb0b322-ea5f-4470-99ae-15b7db787257)

---

On Docker I also got the locale bug, I think the desktop install made some fix for it:

---

![image](https://github.com/fastfetch-cli/fastfetch/assets/61878987/d7220799-ede7-4c39-b369-3d97a00a297d)

**Reply by cassiofb-dev** [reply]
@CarterLi , hi. I made a dockerfile for a musl build. I tried to put this on the workflow but couldn't figure out.
It's on this branch and to test just run ``docker compose up``:
[https://github.com/cassiofb-dev/fastfetch/tree/docker-musl](https://github.com/cassiofb-dev/fastfetch/tree/docker-musl)

This is the compiled binary, it has the same features from @qaqland :
[fastfetch.tar.gz](https://github.com/fastfetch-cli/fastfetch/files/14078128/fastfetch.tar.gz)

Hope this helps someway to setup a musl release.

---
![image](https://github.com/fastfetch-cli/fastfetch/assets/61878987/02ca469a-a2c9-4c43-ba7c-803e2b998bab)

**Reply by qaqland** [reply]
In the past few days, @CarterLi has already done this work!

here it is: https://pkgs.alpinelinux.org/packages?name=fastfetch&branch=edge&repo=&arch=&maintainer=

package is still in edge/testing and you can install it after enable this repository

**Reply by cassiofb-dev** [reply]
Oh, thanks for the update. Thought this was left aside. Next version of Alpine will be epic, probably besides fastfetch we will get plasma 6!

**Comment by qaqland** [comment]
fastfetch has some 3rd part source code or it is easy to port to alpine's

不带第三方代码应该很简单就能提交吧

**Comment by qaqland** [comment]
I wirte a APKBUILD and build fastfetch with almost all dependences:

```
-- The C compiler identification is GNU 13.2.1
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working C compiler: /usr/bin/gcc - skipped
-- Detecting C compile features
-- Detecting C compile features - done
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD - Success
-- Found Threads: TRUE
-- Found PkgConfig: /usr/bin/pkg-config (found version "2.1.0")
-- Build type: None
-- Threads type: pthread
-- The CXX compiler identification is GNU 13.2.1
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: /usr/bin/g++ - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Enabling LTO
-- Target dirs: ROOT="" USR="/usr" HOME="/home" ETC="/etc"
-- Found Python: /usr/bin/python3.11 (found version "3.11.6") found components: Interpreter
-- Enabling DirectX headers for WSL
-- Looking for wcwidth
-- Looking for wcwidth - found
-- Library: found LIBPCI
-- Library: found VULKAN
-- Library: found WAYLAND
-- Library: found XCB_RANDR
-- Library: found XCB
-- Library: missing: XRANDR
-- Library: found X11
-- Library: found DRM
-- Library: found GIO
-- Library: found DCONF
-- Library: found DBUS
-- Library: found XFCONF
-- Library: found SQLITE3
-- Library: missing: RPM
-- Library: found IMAGEMAGICK7
-- Library: missing: IMAGEMAGICK6
-- Library: found ZLIB
-- Library: found CHAFA
-- Library: found EGL
-- Library: missing: GLX
-- Library: found OSMESA
-- Library: found OPENCL
-- Library: found LIBNM
-- Library: found PULSE
-- Library: found DDCUTIL
-- Library: missing: DIRECTX_HEADERS
-- Configuring done (2.2s)
-- Generating done (0.0s)
```

Build success but in my test, "Locale" got **bug**

![image](https://github.com/fastfetch-cli/fastfetch/assets/62464571/e12b6f90-ea29-4544-a42d-3a4e054aa6f2)

**Reply by CarterLi** [reply]
You might set `LANG=C.UTF-8` and `LC_*=C`. what does command `locale` print?

**Reply by qaqland** [reply]
Musl does not support locales
